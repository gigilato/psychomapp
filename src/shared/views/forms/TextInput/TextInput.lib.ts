import { ForwardedRef, useCallback, useImperativeHandle, useRef, useState } from 'react'
import { NativeSyntheticEvent, TextInput, TextInputFocusEventData } from 'react-native'

import { ITextInputProps } from './TextInput.props'

export const useTextInput = (
  ref: ForwardedRef<TextInput>,
  {
    onBlur,
    onFocus,
    onClear,
    error,
  }: Pick<ITextInputProps, 'onFocus' | 'onBlur' | 'onClear' | 'error'>
) => {
  const [isFocused, setIsFocus] = useState(false)
  const innerRef = useRef<TextInput>(null)
  useImperativeHandle(ref, () => (innerRef as any).current)

  const onFieldFocus = useCallback(
    (e: NativeSyntheticEvent<TextInputFocusEventData>) => {
      setIsFocus(true)
      onFocus?.(e)
    },
    [onFocus]
  )

  const onFieldBlur = useCallback(
    (e: NativeSyntheticEvent<TextInputFocusEventData>) => {
      setIsFocus(false)
      onBlur?.(e)
    },
    [onBlur]
  )

  const clearInput = () => {
    innerRef.current?.clear()
    onClear?.()
  }

  const textColor = error ? 'text-danger' : isFocused ? 'text-primary-classic' : 'text-grey-strong'
  const borderColor = error
    ? 'border-danger'
    : isFocused
      ? 'border-primary-classic'
      : 'border-grey-strong'

  return {
    innerRef,
    clearInput,
    textColor,
    borderColor,
    onBlur: onFieldBlur,
    onFocus: onFieldFocus,
  }
}
