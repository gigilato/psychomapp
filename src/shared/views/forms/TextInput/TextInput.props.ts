import { Ref } from 'react'
import { TextInput, TextInputProps } from 'react-native'

import { IconName } from '$atoms'

export interface ITextInputProps extends TextInputProps {
  ID: string
  label?: string
  error?: string
  iconLeft?: IconName
  iconLeftClassName?: string
  onPressIconLeft?: () => void
  iconRight?: IconName
  iconRightClassName?: string
  onPressIconRight?: () => void
  onClear?: () => void
  ref?: Ref<TextInput>
}
