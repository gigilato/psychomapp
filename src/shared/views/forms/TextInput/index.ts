export * from './TextInput'
export * from './TextInput.props'
