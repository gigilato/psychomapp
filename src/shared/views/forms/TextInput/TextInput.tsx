import clsx from 'clsx'
import { forwardRef, memo, useState } from 'react'
import { TextInput as RNTextInput } from 'react-native'

import { Box, HStack, Text, textVariants } from '$atoms'
import { PressableIcon } from '$molecules/Button'
import { colors } from '$theme'

import { useTextInput } from './TextInput.lib'
import { ITextInputProps } from './TextInput.props'

export const TextInput = memo<ITextInputProps>(
  forwardRef<RNTextInput, ITextInputProps>(
    (
      {
        ID,
        onFocus,
        onBlur,
        iconLeft,
        onPressIconLeft,
        iconRight,
        onPressIconRight,
        onClear,
        error,
        className,
        iconRightClassName,
        iconLeftClassName,
        label,
        ...props
      },
      ref
    ) => {
      const { borderColor, textColor, innerRef, clearInput, ...inputProps } = useTextInput(ref, {
        error,
        onBlur,
        onFocus,
        onClear,
      })

      return (
        <Box className={className}>
          {label ? <Text className={clsx('mb-xs', textColor)}>{label}</Text> : null}
          <HStack
            className={clsx(
              'items-center rounded-s py-s border-[1px] bg-white overflow-hidden',
              borderColor
            )}
          >
            {iconLeft && (
              <PressableIcon
                ID={`${ID}/LeftIcon`}
                onPress={onPressIconLeft}
                icon={iconLeft}
                iconClassName={clsx('ml-xs', textColor, iconLeftClassName)}
              />
            )}
            <Box className="flex-1 px-s ">
              <RNTextInput
                testID={ID}
                ref={innerRef}
                className={clsx('text-[16px] leading-[20px]', textVariants.body)}
                keyboardType="default"
                autoCapitalize="none"
                cursorColor={colors.primary.classic}
                selectionColor={colors.primary.classic}
                {...inputProps}
                {...props}
              />
            </Box>
            {(iconRight || props.value !== '') && (
              <PressableIcon
                ID={`${ID}/RightIcon`}
                onPress={onPressIconRight ? onPressIconRight : clearInput}
                icon={iconRight ?? 'cross'}
                iconClassName={clsx('mr-xs', textColor, iconRightClassName)}
              />
            )}
          </HStack>
          {error ? <Text className={clsx('mt-xs', textColor)}>{error}</Text> : null}
        </Box>
      )
    }
  )
)

export const EmailInput = forwardRef<RNTextInput, ITextInputProps>((props, ref) => (
  <TextInput
    ref={ref}
    keyboardType="email-address"
    autoComplete="email"
    iconRight={props.value === '' ? 'envelope' : undefined}
    {...props}
  />
))

export const PasswordInput = forwardRef<RNTextInput, ITextInputProps>(
  ({ iconRight, ...props }, ref) => {
    const [isPasswordVisible, setIsPasswordVisible] = useState(false)

    return (
      <TextInput
        ref={ref}
        textContentType="password"
        autoComplete="password"
        secureTextEntry={!isPasswordVisible}
        iconRight={props.value === '' ? 'lock' : isPasswordVisible ? 'eye-barred' : 'eye'}
        onPressIconRight={() => setIsPasswordVisible(!isPasswordVisible)}
        {...props}
      />
    )
  }
)
