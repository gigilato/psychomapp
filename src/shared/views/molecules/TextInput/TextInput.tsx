import clsx from 'clsx'
import { forwardRef, memo, useImperativeHandle, useRef } from 'react'
import { TextInput as RNTextInput } from 'react-native'

import { Box, HStack, textVariants } from '$atoms'
import { PressableIcon } from '$molecules/Button'
import { colors } from '$theme'

import { ITextInputProps } from './TextInput.props'

export const TextInput = memo<ITextInputProps>(
  forwardRef<RNTextInput, ITextInputProps>(
    (
      {
        ID,
        iconLeft,
        onPressIconLeft,
        iconRight,
        onPressIconRight,
        onClear,
        error,
        className,
        iconRightClassName,
        iconLeftClassName,
        ...props
      },
      ref
    ) => {
      const innerRef = useRef<RNTextInput>(null)
      useImperativeHandle(ref, () => (innerRef as any).current)

      const textColor = error ? 'text-danger' : 'text-grey-strong'
      const borderColor = error ? 'border-danger' : 'border-grey-strong'

      return (
        <HStack
          className={clsx(
            'items-center rounded-s py-s border-[1px] bg-white-light overflow-hidden',
            borderColor,
            className
          )}
        >
          {iconLeft && (
            <PressableIcon
              ID={`${ID}/LeftIcon`}
              onPress={onPressIconLeft}
              icon={iconLeft}
              iconSize={15}
              iconClassName={clsx('ml-xs', textColor, iconLeftClassName)}
            />
          )}
          <Box className="flex-1 px-s ">
            <RNTextInput
              testID={ID}
              ref={innerRef}
              className={clsx('text-[16px] leading-[20px]', textVariants.body)}
              keyboardType="default"
              autoCapitalize="none"
              cursorColor={colors.primary.classic}
              selectionColor={colors.primary.classic}
              {...props}
            />
          </Box>
          {iconRight && (
            <PressableIcon
              ID={`${ID}/RightIcon`}
              onPress={onPressIconRight}
              icon={iconRight}
              iconSize={15}
              iconClassName={clsx('mr-xs', textColor, iconRightClassName)}
            />
          )}
        </HStack>
      )
    }
  )
)
