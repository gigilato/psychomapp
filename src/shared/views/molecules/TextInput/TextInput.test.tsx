import { render } from '@testing-library/react-native'

import { TextInput } from './TextInput'

describe('TextInput', () => {
  it('renders correctly', async () => {
    const component = render(<TextInput ID="test" />)
    expect(component).toMatchSnapshot()
  })
  it('renders correctly with value', async () => {
    const component = render(<TextInput ID="test" value="test" />)
    expect(component).toMatchSnapshot()
  })

  it('renders error correctly', async () => {
    const component = render(<TextInput ID="test" error="error" />)
    expect(component).toMatchSnapshot()
  })
  it('renders icon right correctly', async () => {
    const component = render(<TextInput ID="test" iconRight="bell" />)
    expect(component).toMatchSnapshot()
  })
  it('renders icon left correctly', async () => {
    const component = render(<TextInput ID="test" iconLeft="bell" />)
    expect(component).toMatchSnapshot()
  })
  it('renders icon right error correctly', async () => {
    const component = render(<TextInput ID="test" iconRight="bell" error="error" />)
    expect(component).toMatchSnapshot()
  })
  it('renders icon left error correctly', async () => {
    const component = render(<TextInput ID="test" iconLeft="bell" error="error" />)
    expect(component).toMatchSnapshot()
  })
})
