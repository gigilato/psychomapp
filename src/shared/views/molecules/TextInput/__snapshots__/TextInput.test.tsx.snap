// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "paddingBottom": 12,
        "paddingTop": 12,
      },
      {
        "borderBottomWidth": 1,
        "borderLeftWidth": 1,
        "borderRightWidth": 1,
        "borderTopWidth": 1,
      },
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "overflow": "hidden",
      },
      {
        "borderBottomColor": "#505050",
        "borderLeftColor": "#505050",
        "borderRightColor": "#505050",
        "borderTopColor": "#505050",
      },
      [
        undefined,
        undefined,
      ],
    ]
  }
>
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "paddingLeft": 12,
          "paddingRight": 12,
        },
        [
          undefined,
          undefined,
        ],
      ]
    }
  >
    <TextInput
      autoCapitalize="none"
      className="text-[16px] leading-[20px] font-[SpaceGrotesk-Regular] text-[16px] leading-[18px]"
      cursorColor="#FFB300"
      keyboardType="default"
      selectionColor="#FFB300"
      testID="test"
    />
  </View>
</View>
`;

exports[`TextInput renders correctly with value 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "paddingBottom": 12,
        "paddingTop": 12,
      },
      {
        "borderBottomWidth": 1,
        "borderLeftWidth": 1,
        "borderRightWidth": 1,
        "borderTopWidth": 1,
      },
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "overflow": "hidden",
      },
      {
        "borderBottomColor": "#505050",
        "borderLeftColor": "#505050",
        "borderRightColor": "#505050",
        "borderTopColor": "#505050",
      },
      [
        undefined,
        undefined,
      ],
    ]
  }
>
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "paddingLeft": 12,
          "paddingRight": 12,
        },
        [
          undefined,
          undefined,
        ],
      ]
    }
  >
    <TextInput
      autoCapitalize="none"
      className="text-[16px] leading-[20px] font-[SpaceGrotesk-Regular] text-[16px] leading-[18px]"
      cursorColor="#FFB300"
      keyboardType="default"
      selectionColor="#FFB300"
      testID="test"
      value="test"
    />
  </View>
</View>
`;

exports[`TextInput renders error correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "paddingBottom": 12,
        "paddingTop": 12,
      },
      {
        "borderBottomWidth": 1,
        "borderLeftWidth": 1,
        "borderRightWidth": 1,
        "borderTopWidth": 1,
      },
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "overflow": "hidden",
      },
      {
        "borderBottomColor": "#AE3232",
        "borderLeftColor": "#AE3232",
        "borderRightColor": "#AE3232",
        "borderTopColor": "#AE3232",
      },
      [
        undefined,
        undefined,
      ],
    ]
  }
>
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "paddingLeft": 12,
          "paddingRight": 12,
        },
        [
          undefined,
          undefined,
        ],
      ]
    }
  >
    <TextInput
      autoCapitalize="none"
      className="text-[16px] leading-[20px] font-[SpaceGrotesk-Regular] text-[16px] leading-[18px]"
      cursorColor="#FFB300"
      keyboardType="default"
      selectionColor="#FFB300"
      testID="test"
    />
  </View>
</View>
`;

exports[`TextInput renders icon left correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "paddingBottom": 12,
        "paddingTop": 12,
      },
      {
        "borderBottomWidth": 1,
        "borderLeftWidth": 1,
        "borderRightWidth": 1,
        "borderTopWidth": 1,
      },
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "overflow": "hidden",
      },
      {
        "borderBottomColor": "#505050",
        "borderLeftColor": "#505050",
        "borderRightColor": "#505050",
        "borderTopColor": "#505050",
      },
      [
        undefined,
        undefined,
      ],
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={5}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        [
          {
            "opacity": undefined,
          },
          undefined,
        ],
      ]
    }
    testID="test/LeftIcon"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 15,
          },
          [
            {
              "marginLeft": 8,
            },
            {
              "color": "#505050",
            },
          ],
          {
            "fontFamily": "feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "paddingLeft": 12,
          "paddingRight": 12,
        },
        [
          undefined,
          undefined,
        ],
      ]
    }
  >
    <TextInput
      autoCapitalize="none"
      className="text-[16px] leading-[20px] font-[SpaceGrotesk-Regular] text-[16px] leading-[18px]"
      cursorColor="#FFB300"
      keyboardType="default"
      selectionColor="#FFB300"
      testID="test"
    />
  </View>
</View>
`;

exports[`TextInput renders icon left error correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "paddingBottom": 12,
        "paddingTop": 12,
      },
      {
        "borderBottomWidth": 1,
        "borderLeftWidth": 1,
        "borderRightWidth": 1,
        "borderTopWidth": 1,
      },
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "overflow": "hidden",
      },
      {
        "borderBottomColor": "#AE3232",
        "borderLeftColor": "#AE3232",
        "borderRightColor": "#AE3232",
        "borderTopColor": "#AE3232",
      },
      [
        undefined,
        undefined,
      ],
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={5}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        [
          {
            "opacity": undefined,
          },
          undefined,
        ],
      ]
    }
    testID="test/LeftIcon"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 15,
          },
          [
            {
              "marginLeft": 8,
            },
            {
              "color": "#AE3232",
            },
          ],
          {
            "fontFamily": "feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "paddingLeft": 12,
          "paddingRight": 12,
        },
        [
          undefined,
          undefined,
        ],
      ]
    }
  >
    <TextInput
      autoCapitalize="none"
      className="text-[16px] leading-[20px] font-[SpaceGrotesk-Regular] text-[16px] leading-[18px]"
      cursorColor="#FFB300"
      keyboardType="default"
      selectionColor="#FFB300"
      testID="test"
    />
  </View>
</View>
`;

exports[`TextInput renders icon right correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "paddingBottom": 12,
        "paddingTop": 12,
      },
      {
        "borderBottomWidth": 1,
        "borderLeftWidth": 1,
        "borderRightWidth": 1,
        "borderTopWidth": 1,
      },
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "overflow": "hidden",
      },
      {
        "borderBottomColor": "#505050",
        "borderLeftColor": "#505050",
        "borderRightColor": "#505050",
        "borderTopColor": "#505050",
      },
      [
        undefined,
        undefined,
      ],
    ]
  }
>
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "paddingLeft": 12,
          "paddingRight": 12,
        },
        [
          undefined,
          undefined,
        ],
      ]
    }
  >
    <TextInput
      autoCapitalize="none"
      className="text-[16px] leading-[20px] font-[SpaceGrotesk-Regular] text-[16px] leading-[18px]"
      cursorColor="#FFB300"
      keyboardType="default"
      selectionColor="#FFB300"
      testID="test"
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={5}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        [
          {
            "opacity": undefined,
          },
          undefined,
        ],
      ]
    }
    testID="test/RightIcon"
  >
    <Text />
  </View>
</View>
`;

exports[`TextInput renders icon right error correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "paddingBottom": 12,
        "paddingTop": 12,
      },
      {
        "borderBottomWidth": 1,
        "borderLeftWidth": 1,
        "borderRightWidth": 1,
        "borderTopWidth": 1,
      },
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "overflow": "hidden",
      },
      {
        "borderBottomColor": "#AE3232",
        "borderLeftColor": "#AE3232",
        "borderRightColor": "#AE3232",
        "borderTopColor": "#AE3232",
      },
      [
        undefined,
        undefined,
      ],
    ]
  }
>
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "paddingLeft": 12,
          "paddingRight": 12,
        },
        [
          undefined,
          undefined,
        ],
      ]
    }
  >
    <TextInput
      autoCapitalize="none"
      className="text-[16px] leading-[20px] font-[SpaceGrotesk-Regular] text-[16px] leading-[18px]"
      cursorColor="#FFB300"
      keyboardType="default"
      selectionColor="#FFB300"
      testID="test"
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={5}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        [
          {
            "opacity": undefined,
          },
          undefined,
        ],
      ]
    }
    testID="test/RightIcon"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 15,
          },
          [
            {
              "marginRight": 8,
            },
            {
              "color": "#AE3232",
            },
          ],
          {
            "fontFamily": "feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
